packages/core/src/VRange.ts

```diff
@@ -95,227 +45,101 @@ export class VRange {
-     * `select(a, RelativePosition.AFTER, b, RelativePosition.BEFORE)` => `a[]b`
-     * `select(a, RelativePosition.BEFORE, b, RelativePosition.AFTER)` => `[ab]`
-     * `select(a, RelativePosition.AFTER, b, RelativePosition.AFTER)` => `a[b]`
-     *
-     * @param tailNode
-     * @param [startPosition] default: `RelativePosition.BEFORE`
-     * @param [headNode] default: `startNode`
-     * @param [endPosition] default: `RelativePosition.AFTER`
+     * @param [edge] range edge on which to collapse
      */
-    select(tailNode: VNode, headNode?: VNode): VRange;
-    select(
-        tailNode: VNode,
-        tailPosition: RelativePosition,
-        headNode: VNode,
-        headPosition: RelativePosition,
-    ): VRange;
-    select(
-        tailNode: VNode,
-        tailPosition: RelativePosition | VNode = RelativePosition.BEFORE,
-        headNode: VNode = tailNode,
-        headPosition: RelativePosition = RelativePosition.AFTER,
-    ): VRange {
-        if (tailPosition instanceof VNode) {
-            headNode = tailPosition;
-            tailPosition = RelativePosition.BEFORE;
-        }
-        if (headPosition === RelativePosition.AFTER) {
-            this._setHead(headNode, headPosition);
-            this._setTail(tailNode, tailPosition);
-        } else {
-            this._setTail(tailNode, tailPosition);
-            this._setHead(headNode, headPosition);
+    collapse(edge = this.start): this {
+        if (edge === this.start) {
+            this.setEnd(edge);
+        } else if (edge === this.end) {
+            this.setStart(edge);
         }
         return this;
     }
     /**
      * Set the range's start point (in traversal order) at the given location,
      * targetting a `reference` VNode and specifying the `position` in reference
-     * to that VNode ('BEFORE', 'AFTER'), like in an `xpath`. If a direction is
-     * specified, the range's start point will be the head of the range.
+     * to that VNode ('BEFORE', 'AFTER'), like in an `xpath.
      * Return self.
      *
      * @param reference
      * @param [position]
-     * @param [direction] default: Direction.FORWARD
      */
-    setStart(reference: VNode, position?: RelativePosition, direction = Direction.FORWARD): VRange {
-        if (direction === Direction.FORWARD) {
-            return this._setTail(reference, position);
+    setStart(reference: VNode, position = RelativePosition.BEFORE): this {
+        reference = reference.firstLeaf();
+        if (!reference.hasChildren() && !reference.atomic) {
+            reference.prepend(this.start);
+        } else if (position === RelativePosition.AFTER && reference !== this.end) {
+            // We check that `reference` isn't `this.end` to avoid a backward
+            // collapsed range.
+            reference.after(this.start);
         } else {
-            return this._setHead(reference, position);
+            reference.before(this.start);
         }
+        return this;
```

Comment:
Why do we need to return `this`? It is strange to mix a jQuery style return function with the rest of the code that is not jQuery like and return void.

Reply:
https://github.com/odoo-dev/jabberwock/issues/71